mvebu: add 6.12 as a testing kernel

stklcode opened the pull request:
Kernel bump from 6.6 to 6.12 for testing.

So far pretty straightforward.

To make the WRT1900/WRT32 targets happy _mwlwifi_ needs a small patch: #18980 (merged and rebased)


Compile tested:
* cortexa9
  * buffalo_ls220de
  * cznic_turris-omnia
  * linksys_wrt1900acs
* cortexa53
  * globalscale_espressobin
* cortexa72
  * globalscale_mochabin

Run tested:
* cortexa9
  * linksys_wrt1900ac-v1 (@anomeome)
  * linksys_wrt1900ac-v2 (@borkra)
  * linksys_wrt1900acs (@stklcode)
  * linksys_wrt3200acm (@JamesWWesby)
* cortexa72
  * mikrotik_rb5009 (@sppmasterspp)

----

![wrt1900acs-6 12 31](https://github.com/user-attachments/assets/f1fee638-b907-4c35-a605-8d45dd27d0fc)

stklcode committed:
kernel/mvebu: Create kernel files for v6.12 (from v6.6)

This is an automatically generated commit.

When doing `git bisect`, consider `git bisect --skip`.

Signed-off-by: Stefan Kalscheuer <stefan@stklcode.de>
Link: https://github.com/openwrt/openwrt/pull/18975
Signed-off-by: Hauke Mehrtens <hauke@hauke-m.de>

stklcode committed:
kernel/mvebu: Restore kernel files for v6.6

This is an automatically generated commit which aids following Kernel patch
history, as git will see the move and copy as a rename thus defeating the
purpose.

For the original discussion see:
https://lists.openwrt.org/pipermail/openwrt-devel/2023-October/041673.html

Signed-off-by: Stefan Kalscheuer <stefan@stklcode.de>
Link: https://github.com/openwrt/openwrt/pull/18975
Signed-off-by: Hauke Mehrtens <hauke@hauke-m.de>

stklcode committed:
mvebu: 6.12: refresh patches

Drop upstreamed patches:
0001-v6.16-pinctrl-armada-37xx-use-correct-OUTPUT_VAL-register-.patch
0002-v6.16-pinctrl-armada-37xx-set-GPIO-output-value-before-set.patch
820-v6.11-01-dt-bindings-firmware-add-cznic-turris-omnia-mcu-bind.patch
820-v6.11-02-platform-cznic-Add-preliminary-support-for-Turris-Om.patch
820-v6.11-03-platform-cznic-turris-omnia-mcu-Add-support-for-MCU-.patch
820-v6.11-04-platform-cznic-turris-omnia-mcu-Add-support-for-powe.patch
820-v6.11-05-platform-cznic-turris-omnia-mcu-Add-support-for-MCU-.patch
820-v6.11-06-platform-cznic-turris-omnia-mcu-Add-support-for-MCU-.patch
820-v6.11-07-ARM-dts-turris-omnia-Add-MCU-system-controller-node.patch
820-v6.11-08-ARM-dts-turris-omnia-Add-GPIO-key-node-for-front-but.patch
820-v6.11-09-platform-cznic-turris-omnia-mcu-Depend-on-OF.patch
820-v6.11-10-platform-cznic-turris-omnia-mcu-Depend-on-WATCHDOG.patch
820-v6.11-11-platform-cznic-turris-omnia-mcu-fix-Kconfig-dependen.patch

Manually refreshed:
310-linksys-use-eth0-as-cpu-port.patch
350-drivers-thermal-step_wise-add-support-for-hysteresis.patch
901-dt-bindings-Add-IEI-vendor-prefix-and-IEI-WT61P803-P.patch
902-drivers-mfd-Add-a-driver-for-IEI-WT61P803-PUZZLE-MCU.patch
910-drivers-leds-wt61p803-puzzle-improvements.patch

All other patches automatically refreshed.

Signed-off-by: Stefan Kalscheuer <stefan@stklcode.de>
Link: https://github.com/openwrt/openwrt/pull/18975
Signed-off-by: Hauke Mehrtens <hauke@hauke-m.de>

stklcode committed:
mvebu: 6.12: refresh kernel config

Add new and remove obsolete symbols for Kernel 6.12

Signed-off-by: Stefan Kalscheuer <stefan@stklcode.de>
Link: https://github.com/openwrt/openwrt/pull/18975
Signed-off-by: Hauke Mehrtens <hauke@hauke-m.de>

stklcode committed:
mvebu: add 6.12 testing kernel

Add 6.12 testing kernel for mvebu target.

Signed-off-by: Stefan Kalscheuer <stefan@stklcode.de>
Link: https://github.com/openwrt/openwrt/pull/18975
Signed-off-by: Hauke Mehrtens <hauke@hauke-m.de>

anomeome commented:
Rebase required?

stklcode commented:
Yes, rebased onto the 6.12.31 bump. Also missed a few config symbols. Let's see what breaks in the second run...

anomeome commented:
Not sure what is happening with this PR. Fails to apply with:
```
|diff --git a/target/linux/mvebu/patches-6.12/100-aardvark-workaround-PCIe.patch b/target/linux/mvebu/patches-6.12/100-aardvark-workaround-PCIe.patch
|index 4936f6ad16c0e4..b2282a3684f49d 100644
|--- a/target/linux/mvebu/patches-6.12/100-aardvark-workaround-PCIe.patch
|+++ b/target/linux/mvebu/patches-6.12/100-aardvark-workaround-PCIe.patch
--------------------------
File to patch: 
```

and in the dir :

`
target/linux/mvebu/patches-6.12/100-aardvark-workaround-PCIe.patch.ooFaT62

`

stklcode commented:
Hm, never stumbled across the 100 patch. ran quite a few builds and frefresh cycles over the last days... Does a clean workspace or at least clean target directory help?

----

I'd also like to clean up the `files-6.12` which contains several DTS that are also present upstream. Same holds for 6.6, so not a new issue.
Unfortunately I do not have any of these devices available for testing...

All present upstream with different amounts of changes:
* arch/arm/boot/dts/marvell/armada-370-c200-v2.dts
* arch/arm64/boot/dts/marvell/armada-3720-eDPU.dts
* arch/arm64/boot/dts/marvell/armada-3720-espressobin-ultra.dts
* arch/arm64/boot/dts/marvell/armada-3720-gl-mv1000.dts
* arch/arm64/boot/dts/marvell/armada-3720-uDPU.dts
* arch/arm64/boot/dts/marvell/armada-3720-uDPU.dtsi

stklcode commented:
First run-test on my spare WRT1900ACS was successful. One kernel warning about mtd device 'devinfo' being NAND, but I didn't check if that is new or pre-existing. Device runs fine anyway.

stklcode commented:
Is the `patch` approach supposed to be working well with the copy/restore hack in the first 2 commits (created with `./scripts/kernel_bump.sh` ?
Will try to investigate tomorrow. Neither my local machine (Fedora) nor my CI server (buildworker container) seemed to complain, but all these builds were working on the git tree. Maybe `git am zPatch/18975.patch` could do the trick... 🤔 (or I messed up something, we will seee)

anomeome commented:
git methodology works, interesting that script breaks patch in that fashion, fist time I've seen that behaviour.

compile / test : mamba

dmesg looks fine, kernel on that target device is now:
> 4161719 Jun  1 14:46  linksys_wrt1900ac-v1-kernel.bin

just about at the limit.

anomeome commented:
Enabling  STP on bridge in network config:

``` 
config device                             
        option type 'bridge'              
        option stp '1'                    
```
yields bootlog error:

> br-lan: failed (err=-34) to set attribute (id=6)

not sure if this is new.

borkra commented:
sata not working on cobra:
```
Sat Jun  7 04:31:56 2025 kern.warn kernel: [    6.936528] ata2: link is slow to respond, please be patient (ready=0)
Sat Jun  7 04:31:56 2025 kern.info kernel: [   11.616528] ata2: found unknown device (class 0)
Sat Jun  7 04:31:56 2025 kern.err kernel: [   11.776527] ata2: softreset failed (device not ready)
Sat Jun  7 04:31:56 2025 kern.warn kernel: [   17.136527] ata2: link is slow to respond, please be patient (ready=0)
Sat Jun  7 04:31:56 2025 kern.info kernel: [   21.816527] ata2: found unknown device (class 0)
Sat Jun  7 04:31:56 2025 kern.err kernel: [   21.976527] ata2: softreset failed (device not ready)
Sat Jun  7 04:31:56 2025 kern.warn kernel: [   27.336527] ata2: link is slow to respond, please be patient (ready=0)
Sat Jun  7 04:31:56 2025 kern.info kernel: [   32.316527] ata2: found unknown device (class 0)
Sat Jun  7 04:31:56 2025 kern.warn kernel: [   37.516527] ata2: link is slow to respond, please be patient (ready=0)
Sat Jun  7 04:31:56 2025 kern.err kernel: [   57.016526] ata2: softreset failed (device not ready)
Sat Jun  7 04:31:56 2025 kern.warn kernel: [   57.021600] ata2: limiting SATA link speed to 3.0 Gbps
Sat Jun  7 04:31:56 2025 kern.info kernel: [   62.086533] ata2: found unknown device (class 0)
Sat Jun  7 04:31:56 2025 kern.err kernel: [   62.246532] ata2: softreset failed (device not ready)
Sat Jun  7 04:31:56 2025 kern.warn kernel: [   62.251606] ata2: softreset failed
Sat Jun  7 04:31:56 2025 kern.err kernel: [   62.255021] ata2: reset failed, giving up
```

borkra commented:
Works on linksys_wrt1900acv2

sppmasterspp commented:
I've tried to compile for Mikrotik RB5009 but got this error.

```
Applying /home/debian/openwrt-RB5009UG/target/linux/mvebu/patches-6.12/302-add_powertables.patch using plaintext: 
patching file arch/arm/boot/dts/marvell/armada-385-linksys.dtsi
Hunk #1 succeeded at 212 (offset -2 lines).
patching file arch/arm/boot/dts/marvell/armada-385-linksys-caiman.dts
patching file arch/arm/boot/dts/marvell/armada-385-linksys-cobra.dts
patching file arch/arm/boot/dts/marvell/armada-385-linksys-shelby.dts
patching file arch/arm/boot/dts/marvell/armada-385-linksys-rango.dts
patching file arch/arm/boot/dts/marvell/armada-xp-linksys-mamba.dts

Applying /home/debian/openwrt-RB5009UG/target/linux/mvebu/patches-6.12/304-revert_i2c_delay.patch using plaintext: 
patching file arch/arm/boot/dts/marvell/armada-xp.dtsi

Applying /home/debian/openwrt-RB5009UG/target/linux/mvebu/patches-6.12/305-armada-385-rd-mtd-partitions.patch using plaintext: 
patching file arch/arm/boot/dts/marvell/armada-388-rd.dts

Applying /home/debian/openwrt-RB5009UG/target/linux/mvebu/patches-6.12/306-ARM-mvebu-385-ap-Add-partitions.patch using plaintext: 
patching file arch/arm/boot/dts/marvell/armada-385-db-ap.dts

Applying /home/debian/openwrt-RB5009UG/target/linux/mvebu/patches-6.12/307-armada-xp-linksys-mamba-broken-idle.patch using plaintext: 
patching file arch/arm/boot/dts/marvell/armada-xp-linksys-mamba.dts
Hunk #1 succeeded at 481 (offset -2 lines).

Applying /home/debian/openwrt-RB5009UG/target/linux/mvebu/patches-6.12/308-armada-xp-linksys-mamba-wan.patch using plaintext: 
patching file arch/arm/boot/dts/marvell/armada-xp-linksys-mamba.dts
Hunk #1 succeeded at 383 with fuzz 2 (offset -2 lines).

Applying /home/debian/openwrt-RB5009UG/target/linux/mvebu/patches-6.12/309-linksys-status-led.patch using plaintext: 
patching file arch/arm/boot/dts/marvell/armada-385-linksys.dtsi
patching file arch/arm/boot/dts/marvell/armada-xp-linksys-mamba.dts

Applying /home/debian/openwrt-RB5009UG/target/linux/mvebu/patches-6.12/310-linksys-use-eth0-as-cpu-port.patch using plaintext: 
patching file arch/arm/boot/dts/marvell/armada-385-linksys.dtsi
Hunk #2 FAILED at 200.
1 out of 2 hunks FAILED -- saving rejects to file arch/arm/boot/dts/marvell/armada-385-linksys.dtsi.rej
Patch failed!  Please fix /home/debian/openwrt-RB5009UG/target/linux/mvebu/patches-6.12/310-linksys-use-eth0-as-cpu-port.patch!
make[4]: *** [Makefile:29: /home/debian/openwrt-RB5009UG/build_dir/target-aarch64_cortex-a72_musl/linux-mvebu_cortexa72/linux-6.12.32/.prepared_6c4a1dea9ef3c35f79d1b0231186fce4] Error 1
make[4]: Leaving directory '/home/debian/openwrt-RB5009UG/target/linux/mvebu'
make[3]: *** [Makefile:12: compile] Error 2
make[3]: Leaving directory '/home/debian/openwrt-RB5009UG/target/linux'
time: target/linux/compile#14.89#41.60#63.03
    ERROR: target/linux failed to build.
make[2]: *** [target/Makefile:32: target/linux/compile] Error 1
make[2]: Leaving directory '/home/debian/openwrt-RB5009UG'
make[1]: *** [target/Makefile:25: /home/debian/openwrt-RB5009UG/staging_dir/target-aarch64_cortex-a72_musl/stamp/.target_compile] Error 2
make[1]: Leaving directory '/home/debian/openwrt-RB5009UG'
make: *** [/home/debian/openwrt-RB5009UG/include/toplevel.mk:233: world] Error 2
```

sppmasterspp commented:
> and refreshed patches. No conflicts, no changes required ...

I apply the patch with `patch -p 1 -i 18975.patch` in my builddir? Is it OK because I get prompts like below?
```
File to patch: 
Skip this patch? [y] 
Skipping patch.
3 out of 3 hunks ignored
can't find file to patch at input line 17567
Perhaps you used the wrong -p or --strip option?
The text leading up to this was:
--------------------------
|diff --git a/target/linux/mvebu/patches-6.12/300-mvebu-Mangle-bootloader-s-kernel-arguments.patch b/target/linux/mvebu/patches-6.12/300-mvebu-Mangle-bootloader-s-kernel-arguments.patch
|index 8f5b9dee405df7..2d01b48f69d72c 100644
|--- a/target/linux/mvebu/patches-6.12/300-mvebu-Mangle-bootloader-s-kernel-arguments.patch
|+++ b/target/linux/mvebu/patches-6.12/300-mvebu-Mangle-bootloader-s-kernel-arguments.patch
```

stklcode commented:
I recommend applying the PR patch with `git am` instead, seems `patch -p1` doesn’t like the copy/restore hack from the first 2 commits in a single patch, so the following changes, i.e. required changes to patches and config, get messed up. (we already ran into this issue some comments above)

```
curl https://github.com/openwrt/openwrt/pull/18975.patch | git am
```
or something equivalent with tools/files you prefer typically works fine.

borkra commented:
This is command set to checkout pull requests:
'''
git fetch pull/XXXXX/head:YYY
git merge YYYY
'''
XXXXX - is pull request number
YYYY - is local branch name of your choice

stklcode commented:
Rebased on top of e67b70b8f40cc2ec40c0d3924909ddf81c9beee6 with Kernel 6.12.33.

Dropped 2 more patches that have been backported:
* 0001-v6.16-pinctrl-armada-37xx-use-correct-OUTPUT_VAL-register-.patch (https://github.com/gregkh/linux/commit/f49c751d6000ffb20e798df91e6a010c08318864)
* 0002-v6.16-pinctrl-armada-37xx-set-GPIO-output-value-before-set.patch (https://github.com/gregkh/linux/commit/614456f1a0febc83cd4b902e3f49990c95a39057)
